server/middlewares/fetch-user.js

```diff
@@ -118,29 +137,30 @@ function searchUser(client, query) {
     params.query = { bool: { should, minimum_should_match: 1 } };
   }
 
-  return new Promise((resolve, reject) => {
-    client.post("search/user_reports", params).then(
-      (res = {}) => {
-        const user = res.data && res.data[0];
-        if (!user) return reject(new Error("User not found"));
-        const { id } = user;
-        return Promise.all([
-          client
-            .asUser(id, false)
-            .get(`${id}/segments`)
-            .catch(e =>
-              client.logger.error("fetch.user.segments.error", e.message)),
-          getEventsForUserId(client, id)
-        ]).then((results) => {
-          const [segments = [], events = []] = results;
-          return resolve({ user, segments, events }, reject);
-        });
-      },
-      e => client.logger.error("fetch.user.report.error", e.message)
-    );
-  });
+  return client
+    .post("search/user_reports", params)
+    .then((res = {}) => {
+      const user = res.data && res.data[0];
+      if (!user) throw new Error("User not found");
+      return getEventsAndSegments(client, user);
+    })
+    .catch(e => client.logger.error("fetch.user.segments.error", e.message));
 }
 
+const getSample = source =>
```

Comment:
I would add a docblock explaining what is the expected result of this function 
#flowtypeswillhelp